models/BaseEntity.cfc

```diff
@@ -3140,9 +3267,97 @@ component accessors="true" {
 				variables._nullValues[ alias ] = options.nullValue;
 			}
 		}
+
+		if ( hasParentEntity() ) {
+			explodeParentAttributes();
+		}
+
 		return this;
 	}
 
+	/*=================================
+    =       Subclass Inheritance      =
+    =================================*/
+
+	public boolean function hasParentEntity() {
+		return variables._hasParentEntity;
+	}
+
+	public boolean function isDiscriminatedChild() {
+		return hasParentEntity() && variables._meta.localMetadata.keyExists( "discriminatorValue" );
+	}
+
+	public boolean function isDiscriminatedParent() {
+		return variables._meta.localMetadata.keyExists( "discriminatorColumn" )
+		&& variables._discriminators.len();
+	}
+
+	public function getParentDefinition() {
+		return hasParentEntity() ? variables._parentDefinition : javacast( "null", 0 );
+	}
+
+	public function getDiscriminations() {
+		return variables._cache.getOrSet( "quick-metadata:discriminations-#tableName()#", function() {
```

Comment:
Can you clarify what you mean by `mapping`?    I thought about `entityName()`, but this seemed more likely to prevent a recreation of the cache.  It's probably 6 one/ 1/2 dozen the other, though.

Reply:
`mapping` is the mapping used to create the instance from WireBox.  It is injected in to the metadata.

Reply:
OK. Let me take a look at it.  My only concern would be running in to key name lengths in a non-memory cache provider. It seemed safer to use the table name. It should be able to work 9/10 times.

Reply:
Well, maybe we run with it and come back later if it is a problem.